src/tester/test_unittest.py

```diff
@@ -1,16 +1,20 @@
 import unittest, sys, os, json, warnings, requests, logging
+from unittest.mock import patch, MagicMock, mock_open
```

Comment:
I think overall for the unit test additions, I will have to look deeper into using mock library and how it's used here. But otherwise, it's okay to push to main once we can figure out why that one unit test isn't passing.

Great work!